src/Mealz/AccountingBundle/Controller/CostSheetController.php

```diff
@@ -166,6 +169,15 @@ public function confirmSettlement(
             $profile = $queryResult[0];
             $profile->setSettlementHash(null);
 
+            // Dispatch profile.settlement.confirm event
+            $dispatcher = new EventDispatcher();
+
+            $subscriber = new SettlementSubscriber($this->getDoctrine()->getRepository('MealzMealBundle:Participant'));
```

Comment:
Is there a reason why we are explicitly subscribing a subscriber where we are dispatching the event?

It should work with it.

Reply:
https://symfony.com/doc/current/components/event_dispatcher.html#using-event-subscribers This is what I've got from the doc.

Reply:
This is another reference. https://stackoverflow.com/questions/57540654/symfony-4-custom-event-dispatcher-not-working

Why not write less code and let Symfony handle as many things as possible.

Reply:
Yes, but the example you posted uses a LISTENER and I thought we would like to prefer the SUBSCRIBER method?

Reply:
It is a  subscriber. 

`class TestSubscriber implements EventSubscriberInterface`